plugins/feeds/public/et_open.py

```diff
@@ -0,0 +1,138 @@
+                for ind_cve in self.__extract_cve(rule_suricata.metadata):
+                    ind_suricator.link_to(ind_cve, "affect", "ETOpen")
+
+            if "malware family" in ",".join(rule_suricata.metadata):
+                for in_malware_family in self.__extract_malware_family(
+                    rule_suricata.metadata
+                ):
+                    ind_suricator.link_to(in_malware_family, "affect", "ETOpen")
+
+            tags = self.__extract_tags(rule_suricata.metadata)
+            if tags:
+                ind_suricator.tag(tags)
+            if "mitre_tactic_id" in ",".join(rule_suricata.metadata):
+                for ind_mitre_attack in self.__extract_mitre_attack(
+                    rule_suricata.metadata
+                ):
+                    if ind_mitre_attack:
+                        ind_suricator.link_to(ind_mitre_attack, "affect", "ETOpen")
+
+    def __extract_cve(self, metadata: list):
+        for meta in metadata:
+            if meta.startswith("cve"):
+                _, cve = meta.split(" ")
+                if "_" in cve:
+                    cve = cve.replace("_", "-")
+                ind_cve = entity.Vulnerability.find(name=cve)
+                if not ind_cve:
+                    ind_cve = entity.Vulnerability(name=cve).save()
+                yield ind_cve
+
+    def __extract_malware_family(self, metadata: list):
+        for meta in metadata:
+            if "malware family" in meta:
+                _, malware_family = meta.split(" ")
+                ind_malware_family = entity.Malware.find(name=malware_family)
+                if not ind_malware_family:
+                    ind_malware_family = entity.Malware(name=malware_family).save()
+                yield ind_malware_family
+
+    def __extract_tags(self, metadata: list):
+        tags = []
+        for meta in metadata:
+            if meta.startswith("tag"):
+                _, tag = meta.split(" ")
+                tags.append(tag)
+        return tags
+
+    def __extract_mitre_attack(self, metadata: list):
+        for meta in metadata:
+            if "mitre_tactic_id" in meta:
+                _, mitre_id = meta.split(" ")
+                ind_mitre_attack, nb_ent = entity.Entity.filter(
+                    query_args={"type": entity.EntityType.attack_pattern},
+                    aliases=mitre_id,
+                )
+                if nb_ent != 0:
+                    yield ind_mitre_attack[0]
+
+    def __filter_rule(self, metadata):
+        if not self.last_run:
+            for meta in metadata:
+                if "created_at" in meta:
+                    _, date_create = meta.split(" ")
+                    start_time = yeti_config.get("etopen", "start_time")
+                    if not start_time:
+                        return True
+                    try:
+                        d_start_time = datetime.datetime.strptime(
+                            start_time, "%Y-%m-%d"
+                        )
```

Comment:
This is a big confusing. Why is L118 in the try block? We're parsing a date that we control, so this should work, and if it doesn't we want to error out hard instead of returning `False`. 

Also, you could do `yeti_config.get("etopen", "start_time", "1970-01-01")` and pass the result of that directly to strptime and use it to do the comparison with `date_create`.